plugins/in_opentelemetry/opentelemetry_logs.c

```diff
@@ -585,14 +586,20 @@ static int process_json_payload_log_records_entry(struct flb_opentelemetry *ctx,
         body_object != NULL) {
         result = json_payload_get_wrapped_value(body_object, NULL, &body_type);
 
-        if (result != 0 || body_type == MSGPACK_OBJECT_MAP) {
+        if (result != 0 || 
+            (ctx->logs_body_key == NULL && body_type == MSGPACK_OBJECT_MAP)) {
             flb_log_event_encoder_dynamic_field_reset(&encoder->body);
         }
         else {
+            logs_body_key = ctx->logs_body_key;
+            if (ctx->logs_body_key == NULL) {
+                logs_body_key = "log";
```

Comment:
Even though it's technically not correct (or rather wise) in order to ensue that the compiler does not emit a warning over the loss of the const qualifier please add the appropriate cast here. 

`(char *) "log"`

Reply:
Thanks, added the (unsafe) cast.

Strangely, I didn't see a warning on this line, so tried to reproduce it in a separate simple program, and I see warnings when I do:
```
char *p;
const char a[] = "hello world";
p = a; // warning that assignment discards `const` qualifier
```

however, if I use a fixed string, there's no warning,
```
char *p;
p = "hello world";
```

(even though it's definitely unsafe, if I then later do `p[0] = 'a'`, it seg faults)